envi/archs/aarch64/disasm.py

```diff
@@ -663,28 +921,89 @@ def p_sys(opval, va):
         opcode = INS_SYS
         mnem = 'sys'
         if crn == 0b0111 or crn == 0b1000:
-            sysop_concat = op1+crn+crm+op2
+            # sysop_concat = op1:crn:crm:op2
+            sysop_concat = (op1 << 11) | (crn << 7) | (crm << 3) | op2
+
+            # TODO - Update this for dictionary processing ?
             for bits, mnemonic, ocode in sys_op_table:
                 if sysop_concat == bits:
                     mnem = mnemonic
                     opcode = ocode
                     break
-        olist = (
-            A64ImmOper(op1, 0, S_LSL, va),
-            A64NameOper(crn),
-            A64NameOper(crm),
-            A64ImmOper(op2, 0, S_LSL, va),
-            A64RegOper(rt, va, size=8), #optional operand
-        )
+        
+            # Modify olist based on mnem
+            if mnem == 'at':
+                opcode = INS_AT                
+                olist = (       # AT <at_op>, <Xt>                    
+                    A64NameOper(opcode, ((op1 << 4) | ((crm & 0b1) << 3) | op2)),
+                    A64RegOper(rt, va, size=8)
+                )
+
+            elif mnem == 'dc':
+                opcode = INS_DC
+                olist = (       # DC <dc_op>, <Xt>
```

Comment:
we shouldn't rely on mnemonic.  there has to be a better way, something that doesn't force us into string-comparison here.

Reply:
```suggestion
        
            # Modify olist based on mnem
            if opcode == INS_AT:
                olist = (       # AT <at_op>, <Xt>                    
                    A64NameOper(opcode, ((op1 << 4) | ((crm & 0b1) << 3) | op2)),
                    A64RegOper(rt, va, size=8)
                )

            elif opcode == INS_DC:
                olist = (       # DC <dc_op>, <Xt>
```